generator/views/events_view.py

```diff
@@ -104,3 +102,16 @@ def get_measures(self, dimensions) -> List[Dict[str, str]]:
             )
 
         return measures
+
+    def generate_event_id(self, dimensions: list[dict]) -> Optional[Dict[str, str]]:
+        """Generate the event_id dimension to be used as a primary key for a one to many join."""
+        document_id_field = self.get_document_id(dimensions, "events")
+        event_timestamp = self.select_dimension("event_timestamp", dimensions, "events")
+        event_timestamp_field = event_timestamp["name"] if event_timestamp else None
+        if document_id_field and event_timestamp_field:
+            return {
+                "name": "event_id",
+                "primary_key": "yes",
+                "sql": f"CONCAT(${{{document_id_field}}}, ${{{event_timestamp_field}}})",
```

Comment:
- Rather than `event_timestamp` (which can be the same for multiple events from the same document/ping) I think it would be preferable to have a guaranteed unique ID per event, which I think we can easily achieve using the `UNNEST ... WITH OFFSET AS ...` syntax in `events_unnested`.  I'll file a PR to add a column based on such an offset to `events_unnested` shortly.
- I'd suggest adding a `-` separator between the two values for readability in case us humans want to look at the values.

Reply:
yes offset is a great idea. thanks!

Reply:
#5446